# Exploring Ebay Car Sales Data

The project is aimed to clean and analyze the dataset of used cars from eBay Kleinanzeigen (German eBay website). The dataset includes 50,000 data points from the original dataset uploaded to Kaggle and has such information as name of the car, the barnd and type of the car, price on the ad, date it was listed, postal code of the location and etc.

In [81]:
import pandas as pd
autos = pd.read_csv("autos.csv", encoding= "Latin-1")

In [82]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine_

In [83]:
print(autos.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

In [84]:
print(autos.head())

           dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere 

Based on the info() method we could observe that the dataset. contains 20 columns (most of them are strings. There are certain columns that have less than 50,000 lines, which means they have null values (but none have more that ~20% null values) and they should be observed in more details. These columns include vehicle type, gearbox, fuel type, model and the column stating if the car has the damage which is not yet repaired.
Moreover, there is a column "price" that has the data type as object, instead of the integer or float. Therefore, it has to be modified going forward in the cleaning process.
Based on the observation of the first 5 rows of the dataset, we can see that certain columns are in Gertman and we might need to translate some of them to English. The column with the number of pictures should also be observed in more details as it seems that first 5 rows did not have any photos (and if all values are zero, to delete that column as it brings to value to the analysis).
Moreover, the column names use the camelcase instead of Python's preferred snakecase. 

In [85]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [86]:
autos.columns = ['date_crawled', 
          'name', 
          'seller', 
          'offer_type', 
          'price', 
          'abtest',
          'vehicle_type', 
          'registration_year',
          'gearbox',
          'power_ps',
          'model',
          'odometer',
          'registration_month',
          'fuel_type',
          'brand',
          'unrepaired_damage',
          'ad_created',
          'num_of_pictures',
          'postal_code',
          'last_seen']


autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  num_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00                0   
1         bmw              nein  2016-04-04 00:00:00                0   
2  volkswagen              nein  2016-03-26 00:00:00                0   
3       smart              nein  2016-03-12 00:00:00                0   
4        ford              nein  2016-04-01 00:00:00                0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

Due to the fact that converting column names from camelcase to snakecase cannot be made by replacing spaces with underscores, we renamed each column name individually and reword some of them based on the data disctionary to make it more decriptive. 

In [87]:
autos.describe(include ='all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-21 16:37:21  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  num_of_pictures   postal_code  \
count                 50000          50000.0  50000.000000   
unique                   76              NaN           NaN   
top     2016-04-03 00:00:00              NaN           NaN   
freq                   1946              NaN           NaN   
mean                    NaN              0.0  50813.627300   
std                     NaN              0.0  25779.747957   
min                     NaN              0.0   1067.000000   
25%                     NaN              0.0  30451.000000   
50%                     NaN              0.0  49577.000000   
75%                     NaN              0.0  71540.000000   
max                     NaN              0.0  99998.000000   

                  last_seen  
count                 50000  
unique                39481  
top     2016-04-07 06:17:27  
freq                      8  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

We are using the DataFrame.describe() method to help us explore the dataset. We can observe that the column 'num_of_pictures' might have mostly one value as zero and is a candidate to be dropped. Columns that have less than 50,000 rows might need more investigation to figure out the reason for the null values. 'Seller' and 'offer_type' columns have almost all the same values. Moreover, numeric data stored as text in columns 'price' and 'odometer' need to be cleaned.

In [88]:
autos["num_of_pictures"].value_counts()

0    50000
Name: num_of_pictures, dtype: int64

We used the Series.value_counts() method to have a closer look at the "num_of_pictures" column. As was assumed before, the whole column has zero values. Therefore, it should be dropped as it does not have useful information for the analysis.


In [89]:
autos = autos.drop("num_of_pictures", axis=1)

In [90]:
autos

date_crawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine

Let's now convert 'price' and 'odometer' columns form text to numeric values. In order to do this we will remove any non-numeric characters and convert the column to a numeric dtype.

In [91]:
autos["price"].head()

0    $5,000
1    $8,500
2    $8,990
3    $4,350
4    $1,350
Name: price, dtype: object

In [92]:
autos["price"] = (autos["price"]
                  .str.replace('$','')
                  .str.replace(',','')
                  .astype(int))

In [93]:
autos["odometer"].head()

0    150,000km
1    150,000km
2     70,000km
3     70,000km
4    150,000km
Name: odometer, dtype: object

In [94]:
autos["odometer"] = (autos["odometer"]
                         .str.replace('km','')
                         .str.replace(',','')
                         .astype(int))

autos.rename({"odometer":"odometer_km"}, axis=1,inplace=True)
print(autos["odometer_km"].head())

0    150000
1    150000
2     70000
3     70000
4    150000
Name: odometer_km, dtype: int64


In [95]:
#how many unique values
print(autos["price"].unique().shape)
print(autos["odometer_km"].unique().shape)

(2357,)
(13,)


In [96]:
print(autos["price"].describe())
print(autos["price"].value_counts().head())
print(autos["price"].value_counts().sort_index(ascending=False).head(10))
print(autos["price"].value_counts().sort_index(ascending=True).head(10))

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64
0       1421
500      781
1500     734
2500     643
1000     639
Name: price, dtype: int64
99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
Name: price, dtype: int64
0     1421
1      156
2        3
3        1
5        2
8        1
9        1
10       7
11       2
12       3
Name: price, dtype: int64


In [97]:
print(autos["odometer_km"].describe())
print(autos["odometer_km"].value_counts(dropna=False).head())
print(autos["odometer_km"].value_counts(dropna=False).sort_index(ascending=False).head())
print(autos["odometer_km"].value_counts().sort_index(ascending=True).head())

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64
150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
Name: odometer_km, dtype: int64
150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
Name: odometer_km, dtype: int64
5000     967
10000    264
20000    784
30000    789
40000    819
Name: odometer_km, dtype: int64


Based on the column analysis above, we can see that the minimal value for the 'price' column is zero, which is unrealistic. Moreover, there are 1421 rows that have zero value, therefore, they have to be researched a little bit more. There are also some values with the price being unrealistically high that need to be looked in more details.

In [98]:
print(autos.loc[autos["price"] == 0,"price"])
print(autos.loc[autos["price"] == 99999999, "price"])
print(autos.loc[autos["price"] == 1,"price"])
print(autos.loc[autos["price"] == 27322222,"price"])

27       0
71       0
80       0
87       0
99       0
118      0
146      0
167      0
180      0
226      0
234      0
248      0
259      0
301      0
323      0
327      0
347      0
366      0
389      0
418      0
427      0
430      0
449      0
487      0
502      0
519      0
565      0
593      0
660      0
661      0
        ..
48882    0
48889    0
48891    0
48982    0
48985    0
49043    0
49082    0
49254    0
49301    0
49342    0
49350    0
49404    0
49408    0
49479    0
49496    0
49504    0
49507    0
49525    0
49538    0
49551    0
49614    0
49739    0
49755    0
49793    0
49880    0
49884    0
49943    0
49960    0
49974    0
49984    0
Name: price, Length: 1421, dtype: int64
39705    99999999
Name: price, dtype: int64
55       1
344      1
791      1
981      1
1115     1
1422     1
1561     1
1877     1
1987     1
2442     1
2865     1
2892     1
3062     1
3397     1
3611     1
3679     1
4194     1
4253     1
4323     1
4496     1
4632     1
4787     1
492

In [99]:
print(autos.loc[27])
print(autos.loc[39705])
print(autos.loc[42221])
print(autos.loc[6013])

date_crawled                             2016-03-27 18:45:01
name                  Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE
seller                                                privat
offer_type                                           Angebot
price                                                      0
abtest                                               control
vehicle_type                                             NaN
registration_year                                       2005
gearbox                                                  NaN
power_ps                                                   0
model                                                    NaN
odometer_km                                           150000
registration_month                                         0
fuel_type                                                NaN
brand                                                   ford
unrepaired_damage                                        NaN
ad_created              

After having a deeper research on the outliers (0 and 99,999,999) we can see that they were created by mistake. Therefore, they are to be removed. Some of other prices also could be seen as quite low or too high (like $1 or $10,000,000). However, there is a possibility that $1 cars were just to be given away on the website. While the more expensive price is associated with the luxury car model and the rare year.

In [100]:
autos[autos["price"].between(0,99999999, inclusive=False )]

date_crawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
30     2016-03-14 11:47:31                         Peugeot_206_Unfallfahrzeug   
...                    ...                                                ...   
49968  2016-04-01 17:49:15                  Mercedes_Benz_190_D_2.5_Automatik   
49969  2016-03-17 18:49:02             Nissan_X_Trail_2.2_dCi_4x4_Sport_m.AHZ   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine

After selecting the range between zero and 99,999,999 (not including), we have 48,578 rows left.

# Exploring the date columns

In [101]:
#Extracting just the date values
autos["date_crawled"] = autos["date_crawled"].str.split().str[0]
autos["ad_created"] = autos["ad_created"].str.split().str[0]
autos["last_seen"] = autos["last_seen"].str.split().str[0]

print(autos["date_crawled"].value_counts(normalize=True, dropna=False)
      .sort_index(ascending=True)[:10],"\n")

print(autos["ad_created"].value_counts(normalize=True, dropna=False)
      .sort_index(ascending=True)[:10])
print()
print(autos["last_seen"].value_counts(normalize=True, dropna=False)
      .sort_index(ascending=True)[:10])

2016-03-05    0.02538
2016-03-06    0.01394
2016-03-07    0.03596
2016-03-08    0.03330
2016-03-09    0.03322
2016-03-10    0.03212
2016-03-11    0.03248
2016-03-12    0.03678
2016-03-13    0.01556
2016-03-14    0.03662
Name: date_crawled, dtype: float64 

2015-06-11    0.00002
2015-08-10    0.00002
2015-09-09    0.00002
2015-11-10    0.00002
2015-12-05    0.00002
2015-12-30    0.00002
2016-01-03    0.00002
2016-01-07    0.00002
2016-01-10    0.00004
2016-01-13    0.00002
Name: ad_created, dtype: float64

2016-03-05    0.00108
2016-03-06    0.00442
2016-03-07    0.00536
2016-03-08    0.00760
2016-03-09    0.00986
2016-03-10    0.01076
2016-03-11    0.01252
2016-03-12    0.02382
2016-03-13    0.00898
2016-03-14    0.01280
Name: last_seen, dtype: float64


After analysing the 'date_crawled' and 'last_seen' columns we can observe that the vast majority was created in March, 2016. This could be expalined by the fast that these columns were added by the crawler.

In [102]:
autos["registration_year"].describe()

count    50000.000000
mean      2005.073280
std        105.712813
min       1000.000000
25%       1999.000000
50%       2003.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

Using the Series.describe() method to analyse the 'registration_year' distribution shows us that the data has some odd values like 1000 or 9999, which are unrealistic years for our dataset. 

In [103]:
autos[autos["registration_year"].between(1000,1900, inclusive=False)]

date_crawled                                              name  seller  \
10556   2016-04-01                                        UNFAL_Auto  privat   
24511   2016-03-17                       Trabant__wartburg__Ostalgie  privat   
32585   2016-04-02                                        UNFAL_Auto  privat   
35238   2016-03-26  Suche_Skoda_Fabia____Skoda_Fabia_Combi_mit_Klima  privat   
49283   2016-03-15                                        Citroen_HY  privat   

      offer_type  price   abtest vehicle_type  registration_year gearbox  \
10556    Angebot    450  control          NaN               1800     NaN   
24511    Angebot    490  control          NaN               1111     NaN   
32585    Angebot    450  control          NaN               1800     NaN   
35238    Angebot      0  control          NaN               1500     NaN   
49283    Angebot   7750  control          NaN               1001     NaN   

       power_ps   model  odometer_km  registration_month fuel_type  \
10556      1800     NaN         5000                   2       NaN   
24511         0     NaN         5000                   0       NaN   
32585      1800     NaN         5000                   2       NaN   
35238         0     NaN         5000                   0    benzin   
49283         0  andere         5000                   0       NaN   

            brand unrepaired_damage  ad_created  postal_code   last_seen  
10556  mitsubishi              nein  2016-04-01        63322  2016-04-01  
24511     trabant               NaN  2016-03-17        16818  2016-04-07  
32585  mitsubishi              nein  2016-04-02        63322  2016-04-04  
35238       skoda               NaN  2016-03-26        15517  2016-04-04  
49283     citroen               NaN  2016-03-15        66706  2016-04-06

In [104]:
autos[autos["registration_year"].between(2016,9999, inclusive=False)]


date_crawled                                               name  seller  \
10      2016-03-15                       VW_Golf_Tuning_in_siber/grau  privat   
55      2016-03-07                     Mercedes_E320_AMG_zu_Tauschen!  privat   
65      2016-04-04                      Ford_Fiesta_zum_ausschlachten  privat   
68      2016-04-03  Mini_cooper_s_clubman_/vollausstattung_/_Navi/...  privat   
84      2016-03-27                                     Renault_twingo  privat   
113     2016-04-03                              Golf_4_Anfaenger_auto  privat   
164     2016-03-13  Opel_Meriva__nur_76000_Km__unfallfrei__scheckh...  privat   
197     2016-04-05                              VW_Polo_9N_an_Bastler  privat   
253     2016-03-27             Ford_mondeo_Gas_anlage_mit_TÜV_04.2017  privat   
348     2016-03-17  VW_Beetle_1.8Turbo_mit_Vollausstattung_und_seh...  privat   
390     2016-03-25   Fiat_Bertone_X_1_9__X_1/9__X19__X_19__X1_9__X_19  privat   
438     2016-03-10                VW_Golf_5_V_1.6_2004_Klima_Tempomat  privat   
453     2016-03-28                                         Armee_Jeep  privat   
454     2016-03-25    Ford_Fiesta_Servo_D3_Kat_86€_Steuern_65.000_tkm  privat   
457     2016-03-15  Mercedes_Benz_W203_C200_Kombi_Kompressor_c_kla...  privat   
477     2016-03-31                                         Citroën_C4  privat   
545     2016-03-22                                      Bmw_x5_/__E53  privat   
548     2016-03-10            Peugeot_206_1.9_Hdi_4_Tueren_Tuev_02.17  privat   
557     2016-03-16                           Smart_Fortwo_Passion_MHD  privat   
568     2016-03-18             PRIMA_ANFÄNGERAUTO_smart_fortwo_cabrio  privat   
577     2016-04-03                                  Volkswagen_Golf_3  privat   
582     2016-03-31                                    V10_TDI_TOUAREG  privat   
654     2016-04-01              Ich_verkaufe_meinen_gepflegten_daewoo  privat   
740     2016-04-02                            Seat_Leon_zu_verkaufen!  privat   
754     2016-03-26                         Ford_Ka_Kleinwagen_BJ_2008  privat   
758     2016-03-12                            Honda_Civic_mit_TÜV_!!!  privat   
765     2016-03-26               BMW_3er_Limousine_mit_1_Jahr_TÜV!!!!  privat   
802     2016-03-19                 Lada_mit_wenig_km_neuem_Tuev_bj_08  privat   
871     2016-04-04                                   Citroen_C2_60_PS  privat   
889     2016-03-12                          Verkaufe_Opel_Corsa_C_1_2  privat   
...            ...                                                ...     ...   
49153   2016-03-12                                        Corsa_c20xe  privat   
49154   2016-03-24                    Auto_mit_TÜV_bis_September_2017  privat   
49178   2016-03-24       Mazda_323_F_TÜV__neu_2te_Hand_viele_Neuteile  privat   
49185   2016-03-28              Alfa_Romeo_156_Sportwagon_1_8_T_Spark  privat   
49218   2016-04-01          Ford_Fiesta_1.3_4/5_Tuerer_16_Monate_Tuev  privat   
49245   2016-03-26                         Audi_A3_1.8T_tausch_moegl.  privat   
49259   2016-03-19                  Volkswagen_Lupo_TÜV_06.2017_Klima  privat   
49261   2016-03-10                                        Toyota_aygo  privat   
49262   2016-03-26                                        Opel_vectra  privat   
49266   2016-03-10  Toyota_Corolla_Verso_7_Sitzer_104.000km_2_2_Di...  privat   
49342   2016-03-09  Audi_80_b4_S_Color_und_BMW_318ti_compact_Tausc...  privat   
49343   2016-03-12                                   bmw_320i_touring  privat   
49347   2016-03-21                         Honda_civic_mit_TÜV_3/2017  privat   
49354   2016-04-05                        Bmw_e39_523i_mit_neuem_Tuev  privat   
49389   2016-03-10                                         Astra_G_CC  privat   
49411   2016-03-28                            Renault_twingo_Tuev_neu  privat   
49522   2016-03-14               Mazda_Demio_zu_verkaufen!_RESERVIERT  privat   
49557   2016-03-19                   

Due to the fact that the car cannot be registered after the listing was seen, we suppose that a registration year above 2016 would be inaccurate. Therefore, we have checked the number of listings that fall outside the 1900-2016 interval to see if it is safe to remove thise rows entirely. Couting the intervals from 1000 - 1900 and 2016 - 9999 gives us 5 and 1962 rows respectively, which altogether is approximately 4% of the total dataset. In order to make our dataset more clean and accurate, removing those rows should not have any significantly influence.
Therefore, we are keeping the values in the interval between 1900 - 2016.

In [105]:
autos = autos[autos["registration_year"].between(1900,2016, inclusive=True)]

In [106]:
autos

date_crawled                                               name  seller  \
0       2016-03-26                   Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1       2016-04-04         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   
2       2016-03-26                         Volkswagen_Golf_1.6_United  privat   
3       2016-03-12  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...  privat   
4       2016-04-01  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...  privat   
5       2016-03-21  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...  privat   
6       2016-03-20  VW_Golf_III_GT_Special_Electronic_Green_Metall...  privat   
7       2016-03-16                               Golf_IV_1.9_TDI_90PS  privat   
8       2016-03-22                                         Seat_Arosa  privat   
9       2016-03-16          Renault_Megane_Scenic_1.6e_RT_Klimaanlage  privat   
11      2016-03-16                         Mercedes_A140_Motorschaden  privat   
12      2016-03-31  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...  privat   
13      2016-03-23                                 Audi_A3_1.6_tuning  privat   
14      2016-03-23  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...  privat   
15      2016-04-01        Corvette_C3_Coupe_T_Top_Crossfire_Injection  privat   
16      2016-03-16                                Opel_Vectra_B_Kombi  privat   
17      2016-03-29                          Volkswagen_Scirocco_2_G60  privat   
18      2016-03-26               Verkaufen_mein_bmw_e36_320_i_touring  privat   
19      2016-03-17        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018  privat   
20      2016-03-05  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...  privat   
21      2016-03-06                      Porsche_911_Carrera_4S_Cabrio  privat   
22      2016-03-28                               MINI_Cooper_S_Cabrio  privat   
23      2016-03-10   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima  privat   
24      2016-04-03                         BMW_535i_xDrive_Sport_Aut.  privat   
25      2016-03-21  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung  privat   
26      2016-04-03                                Volkswagen_Polo_Fox  privat   
27      2016-03-27             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE  privat   
28      2016-03-19                                      MINI_Cooper_D  privat   
29      2016-04-02  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...  privat   
30      2016-03-14                         Peugeot_206_Unfallfahrzeug  privat   
...            ...                                                ...     ...   
49970   2016-03-21  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...  privat   
49971   2016-03-29                                         W.Lupo_1.0  privat   
49972   2016-03-26          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.  privat   
49973   2016-03-27                   Mercedes_Benz_SLK_200_Kompressor  privat   
49974   2016-03-20  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...  privat   
49975   2016-03-27          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort  privat   
49976   2016-03-19  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...  privat   
49977   2016-03-31                        Mercedes_Benz_C200_Cdi_W203  privat   
49978   2016-04-04                        Mercedes_Benz_E_200_Classic  privat   
49979   2016-03-20                      Volkswagen_Polo_1.6_TDI_Style  privat   
49980   2016-03-12                            Ford_Escort_Turnier_16V  privat   
49981   2016-03-15             Opel_Astra_Kombi_mit_Anhaengerkupplung  privat   
49982   2016-03-29            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm  privat   
49983   2016-03-06                                      Ford_focus_99  privat   
49984   2016-03-31  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...  privat   
49985   2016-04-02                          Verkaufe_meinen_vw_vento!  privat   
49986   2016-04-04  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...  privat   
49987   2016-03-22  Audi_A3_Limousine

In [107]:
print(autos["registration_year"].value_counts(normalize = True, dropna = False))
print(autos["registration_year"].value_counts(normalize = True, dropna = False)
                                .sort_index(ascending=True)[:10],"\n")
print(autos["registration_year"].value_counts(normalize = True, dropna = False)
                                .sort_index(ascending=False)[:10],"\n")




2000    0.069834
2005    0.062776
1999    0.062464
2004    0.056988
2003    0.056779
2006    0.056384
2001    0.056280
2002    0.052740
1998    0.051074
2007    0.047972
2008    0.046452
2009    0.043683
1997    0.042225
2011    0.034022
2010    0.033251
1996    0.030066
2012    0.027546
2016    0.027401
1995    0.027338
2013    0.016782
2014    0.013867
1994    0.013742
1993    0.009265
2015    0.008308
1990    0.008224
1992    0.008141
1991    0.007412
1989    0.003769
1988    0.002957
1985    0.002186
          ...   
1966    0.000458
1977    0.000458
1975    0.000396
1969    0.000396
1965    0.000354
1964    0.000250
1910    0.000187
1963    0.000187
1959    0.000146
1961    0.000125
1956    0.000104
1958    0.000083
1937    0.000083
1962    0.000083
1950    0.000062
1954    0.000042
1941    0.000042
1934    0.000042
1957    0.000042
1951    0.000042
1955    0.000042
1931    0.000021
1953    0.000021
1943    0.000021
1938    0.000021
1939    0.000021
1927    0.000021
1929    0.0000

After analysing the cleaned dataset we can observe that the highest percentage ~7% of the cars were registered in 2000 and the lowest in 1952. The oldest registration year is 1910.

In [108]:
print(autos["brand"].unique(),'\n')
print(autos["brand"].value_counts().head(20))
print(autos["brand"].value_counts().head(20).index)


['peugeot' 'bmw' 'volkswagen' 'smart' 'ford' 'chrysler' 'seat' 'renault'
 'mercedes_benz' 'audi' 'sonstige_autos' 'opel' 'mazda' 'porsche' 'mini'
 'toyota' 'dacia' 'nissan' 'jeep' 'saab' 'volvo' 'mitsubishi' 'jaguar'
 'fiat' 'skoda' 'subaru' 'kia' 'citroen' 'chevrolet' 'hyundai' 'honda'
 'daewoo' 'suzuki' 'trabant' 'land_rover' 'alfa_romeo' 'lada' 'rover'
 'daihatsu' 'lancia'] 

volkswagen        10188
bmw                5284
opel               5195
mercedes_benz      4580
audi               4149
ford               3352
renault            2274
peugeot            1418
fiat               1242
seat                873
skoda               770
mazda               727
nissan              725
citroen             669
smart               668
toyota              599
sonstige_autos      526
hyundai             473
volvo               444
mini                415
Name: brand, dtype: int64
Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat'

In order to explore variations across different car brands, we are exploring unqiue values in the 'brand' column and chooseing the top 20 brands to aggregate by. The top brand in our dataset is Volkswagen and the least out of 20 is Mini.

In [109]:
selected_brands = autos["brand"].value_counts().head(20).index
brand_dict = {}

for b in selected_brands:
    selected_rows = autos[autos["brand"] == b]
    mean_price = selected_rows["price"].mean()
    brand_dict[b] = mean_price
    
print(brand_dict)



{'toyota': 5115.33388981636, 'peugeot': 3039.4682651622, 'fiat': 2711.8011272141707, 'ford': 7263.015811455847, 'mini': 10460.012048192772, 'mercedes_benz': 30317.447816593885, 'bmw': 8334.645155185466, 'volvo': 4757.108108108108, 'volkswagen': 6516.457597173145, 'sonstige_autos': 39621.77946768061, 'mazda': 4010.7716643741405, 'hyundai': 5308.53911205074, 'seat': 4296.492554410081, 'nissan': 4664.891034482758, 'audi': 9093.65003615329, 'skoda': 6334.91948051948, 'opel': 5252.61655437921, 'smart': 3542.706586826347, 'renault': 2395.4164467897976, 'citroen': 44534.79671150971}


In [110]:
for w in sorted(brand_dict,key = brand_dict.get, reverse =True):
    print (w, brand_dict[w])

citroen 44534.79671150971
sonstige_autos 39621.77946768061
mercedes_benz 30317.447816593885
mini 10460.012048192772
audi 9093.65003615329
bmw 8334.645155185466
ford 7263.015811455847
volkswagen 6516.457597173145
skoda 6334.91948051948
hyundai 5308.53911205074
opel 5252.61655437921
toyota 5115.33388981636
volvo 4757.108108108108
nissan 4664.891034482758
seat 4296.492554410081
mazda 4010.7716643741405
smart 3542.706586826347
peugeot 3039.4682651622
fiat 2711.8011272141707
renault 2395.4164467897976


After aggregating the brand data and sorting it in descending order, we can observe that citroen cars have the highest mean price amount our selected brands, while renault has the lowest mean price.

In [111]:
selected_brands = autos["brand"].value_counts().head(6).index
brand_mean_price = {}
brand_mean_mileage = {}

for b in selected_brands:
    selected_rows = autos[autos["brand"] == b]
    mean_price = selected_rows["price"].mean()
    mean_mileage = selected_rows["odometer_km"].mean()
    brand_mean_price[b] = mean_price
    brand_mean_mileage[b] = mean_mileage
    
print(brand_mean_price,'\n')
print(brand_mean_mileage)

{'audi': 9093.65003615329, 'ford': 7263.015811455847, 'mercedes_benz': 30317.447816593885, 'bmw': 8334.645155185466, 'volkswagen': 6516.457597173145, 'opel': 5252.61655437921} 

{'audi': 129287.78018799711, 'ford': 124046.83770883054, 'mercedes_benz': 130860.26200873363, 'bmw': 132434.70855412565, 'volkswagen': 128730.36906164115, 'opel': 129227.14148219442}


In [112]:
#Converting both dictionaries to series objects, using the series constructor
bmp_series = pd.Series(brand_mean_price)
bmm_series = pd.Series(brand_mean_mileage)

#Creating a dataframe from the first series object using the dataframe constructor

df = pd.DataFrame(bmp_series, columns=['mean_price'])


In [113]:
df

mean_price
audi            9093.650036
bmw             8334.645155
ford            7263.015811
mercedes_benz  30317.447817
opel            5252.616554
volkswagen      6516.457597

In [114]:
#Assigning the other series as a new column in the dataframe
df['mean_mileage'] = bmm_series

In [115]:
df

mean_price   mean_mileage
audi            9093.650036  129287.780188
bmw             8334.645155  132434.708554
ford            7263.015811  124046.837709
mercedes_benz  30317.447817  130860.262009
opel            5252.616554  129227.141482
volkswagen      6516.457597  128730.369062

In [118]:
df=df.sort_values("mean_price",ascending=False)
df

mean_price   mean_mileage
mercedes_benz  30317.447817  130860.262009
audi            9093.650036  129287.780188
bmw             8334.645155  132434.708554
ford            7263.015811  124046.837709
volkswagen      6516.457597  128730.369062
opel            5252.616554  129227.141482

After cretating the dataframe containing both mean price and mean mileage figures, we can see that there is no certain link between price and mileage. Futhermore, average price for mercedes seems to be distinct from other car brands. Looking separately to mercedes figures shows that the dataset also includes many limousines that increase the avearge price.